webhook/resources/engine/validator.go

```diff
@@ -68,3 +77,23 @@ func (e *engineValidator) Update(request *admission.Request, oldObj runtime.Obje
 
 	return nil
 }
+
+func (e *engineValidator) validateNumberOfEngines(newEngine *longhorn.Engine, volume *longhorn.Volume) error {
+	volumeEngines, err := e.ds.ListVolumeEnginesUncached(newEngine.Spec.VolumeName)
+	if err != nil {
+		err = errors.Wrap(err, "failed to list engines for volume")
+		return werror.NewInternalError(err.Error())
+	}
+
+	newNumVolumeEngines := len(volumeEngines) + 1
+	if volume.Spec.Migratable && newNumVolumeEngines > 2 {
+		message := fmt.Sprintf("engine creation would result in %d engines for migratable volume", newNumVolumeEngines)
+		return werror.NewInvalidError(message, "spec.volumeName")
+	}
+	if !volume.Spec.Migratable && newNumVolumeEngines > 1 {
+		message := fmt.Sprintf("engine creation would result in %d engines for non-migratable volume", newNumVolumeEngines)
+		return werror.NewInvalidError(message, "spec.volumeName")
```

Comment:
It seems this validation is to check specific conditions instead of a field, so `"spec.volumeName"` should not be required.

Reply:
Good point. Fixed. Thank you